toolchain/check/handle_name.cpp

```diff
@@ -19,10 +21,35 @@ auto HandleMemberAccessExpr(Context& context, Parse::MemberAccessExprId node_id)
   return true;
 }
 
+auto HandlePrefixOperatorStar(Context& context,
+                              Parse::PrefixOperatorStarId node_id) -> bool {
+  auto base_id = context.node_stack().PopExpr();
+  auto type_id =
+      context.GetUnqualifiedType(context.insts().Get(base_id).type_id());
+  // TODO: Check for any facet here, rather than only a type.
+  if (type_id == SemIR::TypeId::TypeType) {
+    CARBON_DIAGNOSTIC(
+        DerefOfType, Note,
+        "To form a pointer type, write the `*` after the pointee type.");
+    auto builder = context.emitter().Build(TokenOnly(node_id), DerefOfType);
+    builder.Note(TokenOnly(node_id), DerefOfType);
+    builder.Emit();
+  }
+  auto dereference_id = PerformPointerDereference(context, node_id, base_id);
+  context.AddInstAndPush({node_id, dereference_id});
+  return true;
+}
+
 auto HandlePointerMemberAccessExpr(Context& context,
                                    Parse::PointerMemberAccessExprId node_id)
     -> bool {
-  return context.TODO(node_id, "HandlePointerMemberAccessExpr");
+  auto [name_node_id, name_id] = context.node_stack().PopNameWithNodeId();
+  auto base_id = context.node_stack().PopExpr();
+  auto dereference_id = PerformPointerDereference(context, node_id, base_id);
+  context.AddInstAndPush({node_id, dereference_id});
+  auto member_id = PerformMemberAccess(context, node_id, base_id, name_id);
+  context.node_stack().Push(node_id, member_id);
```

Comment:
You're pushing `node_id` onto the node stack twice here, once in `AddInstAndPush` and again in `node_stack().Push`. That's responsible for at least one of the two test issues you're seeing. We don't actually want the dereferenced base to be on the stack, because we don't need to pass it to the next check action, so:

```suggestion
  auto deref_base_id = context.AddInst({node_id, dereference_id});
  auto member_id = PerformMemberAccess(context, node_id, deref_base_id, name_id);
  context.node_stack().Push(node_id, member_id);
```

I think the `AddInst` call for the dereference should be in `PerformPointerDereference` rather than here; we typically have check operations create and add instructions themselves. (That way, if they want to create multiple instructions, they don't need to treat the last one differently, and if they end up creating no instructions and just returning an operand, they can do so.)